src/vs/workbench/services/positronNewProject/common/positronNewProjectService.ts

```diff
@@ -533,6 +544,206 @@ export class PositronNewProjectService extends Disposable implements IPositronNe
 
 	//#endregion Extension Tasks
 
+
+	/**
+	 * Centralised validation helper for notebook connection.
+	 * Prefers the active notebook editor if present, otherwise falls back to the only open notebook editor.
+	 * If no notebook editors are open, or no active notebook editor is present when multiple are open, logs and notifies the user.
+	 *
+	 * @returns The validated context required to establish a runtime session, or `undefined` if validation failed.
+	 */
+	private _getNotebookContext(): { model: INotebookTextModel; runtimeId: string } | undefined {
+		const notebookEditors = this._notebookEditorService.listNotebookEditors();
+
+		// Prefer the active notebook editor if available
+		const activeEditor = (this._notebookEditorService as any).activeNotebookEditor as { textModel?: INotebookTextModel } | undefined;
```

Comment:
Should the `INotebookEditorService` interface have `activeNotebookEditor` as a property, so that we don't need to do these type assertions?